### PR TITLE
Feature/397

### LINKED ISSUE #397: Update pulled content management
**Is your enhancement related to a problem? Please describe.**
Include various states for better communicating the current status of distributed content.

**Describe the solution you'd like**
- [ ] in Posts table view, include better details in Distributor icon column, add Pushed/Pulled/Linked/Unlinked in hover details for each item (see: https://cldup.com/kVpFJIy8bM.png)
- [ ] links to linked content should open original in a new tab (see: https://cldup.com/kVpFJIy8bM.png)
- [ ] add `Pulled content` sidebar (see: https://cldup.com/c9gR5SBTPB.png)
- [ ] on clicking `Unlink from Original` show confirmation to ensure user understands the implications of the action (see: https://cldup.com/c9gR5SBTPB.png)
- [ ] add `Unlinked content` sidebar (see: https://cldup.com/JKwubnt5x1.png)
- [ ] on clicking `Restore link to Original` show confirmation to remove ambiguity around data overwrite implications, then go back to `Pulled content` state (see: https://cldup.com/JKwubnt5x1.png)

**Designs**
1. https://cldup.com/kVpFJIy8bM.png
1. https://cldup.com/c9gR5SBTPB.png
1. https://cldup.com/JKwubnt5x1.png

**Describe alternatives you've considered**
none

**Additional context**
Props GH:@oszkarnagy / WP:[@oszkarnagy](https://profiles.wordpress.org/oszkarnagy/) for the UX analysis and recommendation.

### PR DESCRIPTION
**Pull request by roshniahuja**
### Description of the Change

- In Posts table view, include better details in Distributor icon column, add Pushed/Pulled/Linked/Unlinked in hover details for each item (see: https://cldup.com/kVpFJIy8bM.png)

image

- Links to linked content should open original in a new tab (see: https://cldup.com/kVpFJIy8bM.png) add Pulled content sidebar (see: https://cldup.com/c9gR5SBTPB.png)
-  On clicking Unlink from Original show confirmation to ensure user understands the implications of the action (see: https://cldup.com/c9gR5SBTPB.png)
-  Add Unlinked content sidebar (see: https://cldup.com/JKwubnt5x1.png)
-  On clicking Restore link to Original show confirmation to remove ambiguity around data overwrite implications, then go back to Pulled content state (see: https://cldup.com/JKwubnt5x1.png)

Closes #397

### Changelog Entry

> Added - Impvoed Pulled Content Management

### Credits

Props @roshniahuja @faisal-alvi 

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

**Comment by roshniahuja** [comment]
@peterwilsoncc ,

I have to Apply condition based Title for PluginDocumentSettingPanel and for that I had created const and added that const in title as below but somehow it is not working. Can you please check once?

> title={ DistributorTitle }

 https://github.com/10up/distributor/blob/88761bed134235ead796ed9b9fa0e651c4182edb/assets/js/gutenberg-plugin.js#:~:text=)%3B-,/**,%7D%3B,-/**

**Comment by roshniahuja** [comment]
@jeffpaul ,

I have integrated this feature in Modal but Need to setup design so can you please assign one designer for the same?

**Comment by roshniahuja** [comment]
> @peterwilsoncc ,
> 
> I have to Apply condition based Title for PluginDocumentSettingPanel and for that I had created const and added that const in title as below but somehow it is not working. Can you please check once?
> 
> > title={ DistributorTitle }
> 
> https://github.com/10up/distributor/blob/88761bed134235ead796ed9b9fa0e651c4182edb/assets/js/gutenberg-plugin.js#:~:text=)%3B-,/**,%7D%3B,-/**

@peterwilsoncc , You can ignore this as its been resolved.

**Review by ravinderk — COMMENTED** [review]
We can hold the review. I raised a question on issue https://github.com/10up/distributor/issues/397#issuecomment-1582799450

**Comment by faisal-alvi** [comment]
Note:
- The design part is completed.
- Added 3 different icons to show the status of a post. 1. Linked 2. Unlinked 3. Pushed

image
(_image also added to the main description_)

E2E failures should be ignored as they will be addressed in https://github.com/10up/distributor/pull/1242.

File: assets/js/gutenberg-plugin.js [hunk 1]
```diff
@@ -131,14 +132,136 @@ const RenderDistributedTo = () => {
...
+				<span id="distributed-data">
+					{ __( 'Updating the ', 'distributor' ) }
+					<a href={ dtGutenberg.postUrl }>
+						{ __( 'Original Content', 'distributor' ) },
+					</a>
+					{ __(
+						'will update this post automatically.',
+						'distributor'
+					) }
+					,
```
Comment (iamdharmesh) [inline assets/js/gutenberg-plugin.js]:
```suggestion
```

File: assets/js/gutenberg-plugin.js [hunk 2]
```diff
@@ -131,14 +132,136 @@ const RenderDistributedTo = () => {
...
+						),
+						dtGutenberg.syndicationTime,
+						dtGutenberg.originalLocationName
+					) }
+				</span>
+				<span id="distributed-data">
+					{ __( 'Updating the ', 'distributor' ) }
+					<a href={ dtGutenberg.postUrl }>
+						{ __( 'Original Content', 'distributor' ) },
+					</a>
```
Comment (iamdharmesh) [inline assets/js/gutenberg-plugin.js]:
```suggestion
						{ __( 'Original Content', 'distributor' ) }
					</a>
					{ ', ' }
```

Adding extra space here will fix the spacing issue.
image

**Review by iamdharmesh — CHANGES_REQUESTED** [review]
Thanks for working on this @faisal-alvi. Great work here, I have added 2 minor comments. Also, for me the "Unlink" button is disabled and I am also not able to click (X) on the popup. Could you please help to check?

image


Thank you.

**Review by Sidsector9 — DISMISSED** [review]
Tests well 👍 , Approving not merging yet.

@jeffpaul following up on the comment by Ravinder here - https://github.com/10up/distributor/issues/397#issuecomment-1582799450

cc: @dkotter as Jeff is on PTO.

If it's cool with you then it's good to go.

**Comment by dkotter** [comment]
> following up on the comment by Ravinder here - https://github.com/10up/distributor/issues/397#issuecomment-1582799450

I think I agree that these icons could be clearer but I also don't have a great suggestion in mind. I think we're fine to ship this as-is and we can iterate on it later.

That said, I see there's a couple comments by @iamdharmesh that need addressed and also seeing a bunch of eslint issues we should clean up. Once those are taken care of, I'm good to get this merged in

File: includes/syndicated-post-ui.php [hunk 3]
```diff
@@ -79,18 +79,23 @@ function output_distributor_column( $column_name, $post_id ) {
...
+			if ( $connection_map ) {
+				// When a post is pushed from current site or pulled by other sites.
+				echo '<span title="' . esc_attr__( 'Pushed', 'distributor' ) . '" class="dashicons dashicons-admin-page"></span>';
 			} else {
-				echo '<a href="' . esc_url( $post_url ) . '"><img src="' . esc_url( plugins_url( 'assets/img/icon.svg', __DIR__ ) ) . '" alt="' . esc_attr__( 'Linked', 'distributor' ) . '" title="' . esc_attr__( 'Linked', 'distributor' ) . '"></a>';
+				$unlinked = (bool) get_post_meta( $post_id, 'dt_unlinked', true );
+				$post_url = get_post_meta( $post_id, 'dt_original_post_url', true );
+	
+				if ( $unlinked ) {
+					echo '<a href="' . esc_url( $post_url ) . '"><span title="' . esc_attr__( 'Unlinked', 'distributor' ) . '" class="dashicons dashicons-editor-unlink"></span></span></a>';
```
Comment (iamdharmesh) [inline includes/syndicated-post-ui.php]: Maybe this should also open in a new tab like Linked post?
```suggestion
					echo '<a target="_blank" href="' . esc_url( $post_url ) . '"><span title="' . esc_attr__( 'Unlinked', 'distributor' ) . '" class="dashicons dashicons-editor-unlink"></span></span></a>';
```

**Review by iamdharmesh — DISMISSED** [review]
Thanks for the changes @faisal-alvi @Sidsector9. LGTM.

Just added a minor suggestion but nothing blocker.

**Review by iamdharmesh — APPROVED** [review]

**Pull request merged**